### ISSUE TITLE
Bug: Dice So Nice can't display to others for character rolls

### ISSUE THREAD

**Issue by bithir** [pwatson100/symbaroum#190]
Technically, according to the API, if unset, it should default to show to everyone. It does not.
Also, make it configurable

**Comment by JiDW** [comment]
Hey!

I saw this issue on Discord.
The problem is not the "whisper" parameter but the lack of the "synchronize" parameter. Without this parameter, it won't be sent to other players.

Also if you have this possibility, we strongly recommend the "No API" approach to integrate DsN by just adding the roll object to the chat message:

https://gitlab.com/riccisi/foundryvtt-dice-so-nice/-/wikis/Integration

**Comment by bithir** [comment]
Just made a quick check - it looks unlikely we can use the "No API" - our rolls includes multiple separate rolls, so unless dice-so-nice can take an array of roll objects, I do not think it is an option for us.

**Comment by bithir** [comment]
Foundry API does not allow an array of roll objects, so I think we will have to stick to using the API version.

**Comment by JiDW** [comment]
It is easy to create a single roll from an array of rolls in 0.8:

```javascript
 const rolls = []; //array of Roll
 const pool = PoolTerm.fromRolls(rolls);
 roll = Roll.fromTerms([pool]);
```

You can do it in 0.7 too, it's just a bit more lines, that's why this method was added in 0.8

```javascript
let rolls = [];//array of Roll
let mergingPool = new DicePool({rolls:rolls}).evaluate();
roll = Roll.create(mergingPool.formula).evaluate();
roll.terms = [mergingPool]
roll.results = [mergingPool.total];
roll._total = mergingPool.total;
roll._rolled = true;
```

**Comment by bithir** [comment]
Not sure about using the the non-api for the system - all dice are dumped directly at the same time, which I think will make it tricky to understand for the user which dice belongs to which test, unless there is some means to instruct the pacing between displaying the different rolls.

**Comment by bithir** [comment]
Tested with option for merging simultaneous rolls set to off - and it does not do the thing for us.

**Comment by JiDW** [comment]
> 
> 
> Not sure about using the the non-api for the system - all dice are dumped directly at the same time, which I think will make it tricky to understand for the user which dice belongs to which test, unless there is some means to instruct the pacing between displaying the different rolls.

Not with a single chat message no, sorry :( 
Thank you for trying! It's very useful for me to be able to understand cases where the Roll API is needed because of the chatmessage system limitations. Sorry for making you try these for nothing!

**Comment by bithir** [comment]
It was good to see how it worked - if there could be a flag that I could add to indicate that it needs to wait and/or a delay between the roll, then we would use it. Want me to add it as a Feature Request and you can consider if you want to have it implemented?

**Comment by JiDW** [comment]
If you get the chance to post it yourself so you can write down a quick requirement list, that's would be awesome! I sometimes get lost between systems and having some context when I try to tackle a feature request is a godsend

**Comment by NeV3rKilL** [comment]
I modified roll.js to add the "synchronize=true" parameter to  game.dice3d.showForRoll call to fix this.

https://github.com/pwatson100/symbaroum/pull/215

**Comment by bithir** [comment]
Fixed in 3.0.2
